### PR TITLE
mpdecimal: new recipe

### PR DESCRIPTION
**Pull request by hundeboll**
Needed to build python3

### CODE REVIEW

File: recipes/mpdecimal/mpdecimal.inc [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+LICENSE = "BSD-2-Clause"
+HOMEPAGE = "http://www.bytereef.org/mpdecimal/index.html"
+RECIPE_TYPES = "machine native"
+
+inherit c autotools library
+
+DEPENDS = "libm"
+SRC_URI = "http://www.bytereef.org/software/${PN}/releases/${PN}-${PV}.tar.gz"
+
+LIBRARY_NAME = "libmpdecimal"
```
Comment (esben) [inline recipes/mpdecimal/mpdecimal.inc]: Why not set it to "libmpdec" instead?

File: recipes/mpdecimal/mpdecimal.inc [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+LIBRARY_VERSION = "2"
+
+FILES_${PN} += "${libdir}/libmpdec.so.${LIBRARY_VERSION} ${libdir}/libmpdec.so.${PV}"
+FILES_${PN}-dev += "${libdir}/libmpdec.so"
+DEPENDS_${PN} = "libc libm"
+RDEPENDS_${PN} = "libc libm"
+PROVIDES_${PN} = "libmpdec libmpdec-${LIBRARY_VERSION}"
+PROVIDES_${PN}[qa] += "allow-missing-soname:libmpdecimal-2"
+PROVIDES_${PN}[qa] += "allow-missing-soname:libmpdecimal"
```
Comment (esben) [inline recipes/mpdecimal/mpdecimal.inc]: If you change LIBRARY_NAME as suggested above, I think the 3 lines here will not be needed (handled automatically by library.oeclass).

Comment (hundeboll) [inline recipes/mpdecimal/mpdecimal.inc]: How is it handled when the library uses different names for header vs. library?

E.g.
```
/usr/include/mpdecimal.h
/usr/lib/libmpdec.so.2
/usr/lib/.debug/libmpdec.so.2.4.1
/usr/lib/libmpdec.so.2.4.1
```

Comment (hundeboll) [inline recipes/mpdecimal/mpdecimal.inc]: Disregard that. I am reworking the name now..

Comment (esben) [inline recipes/mpdecimal/mpdecimal.inc]: If you haven't figured it out, you can always add files to FILES_${PN} and FILES_${PN}-dev as you like.

**Comment by esben** [comment]
Much better, LGTM :+1:

**Pull request merged**